jobs/github/scripts/snippet_build_basic-make-check.sh

```diff
@@ -2,11 +2,18 @@
 
     NEEDS_MGO=$(echo "${NEEDS_MGO}" | tr '[:upper:]' '[:lower:]')
     if [ "${NEEDS_MGO}" = "true" ]; then
-      echo "Installing juju-db snap."
+      action=$(snap info juju-db | grep -q "installed" || echo "install")
+      if [ "$action" == "" ]; then
+              action="refresh"
+      fi
+
       # when running inside a privileged container, snapd fails because udevd isn't
       # running, but on the second occurance it is.
       # see: https://github.com/lxc/lxd/issues/4308
-      sudo snap refresh juju-db --channel=4.4 2> /dev/null || sudo snap install juju-db --channel=4.4
+      for i in {1..3}; do
+        echo "${action}ing juju-db snap. Attempt ${i}"
+        sudo snap $action juju-db --channel=4.4 && echo $action"ed:" && which juju-db.mongo && juju-db.mongo --version && break || sleep 10
```

Comment:
In general, I'm not a big fan of deeply nested lines like this (they can be hard to maintain, etc)
Certainly since if everything works you 'break' there is no need to have `|| sleep 10` on the end, because you can just put `sleep 10` on the next line and it will be skipped. However, if we instead just did:

```suggestion
        if sudo snap $action juju-db --channel=4.4; then
          echo $action"ed:"
          # If juju-db is correctly installed, report the version and stop looping
          which juju-db.mongod && juju-db.mongod --version && break
       fi
        sleep 10
```

Also note that we really care about the `mongod` (server) version, not the `mongo` (client) version. Though in my testing they return nearly identical content (the first line about `MongoDB shell` vs `db version` is the main difference)

Reply:
Great points, @jameinel, thanks! I'll make the changes asap :+1: